pkg/vsphere/tasks/waiter.go

```diff
@@ -27,12 +28,47 @@ import (
 	"github.com/vmware/govmomi/vim25/progress"
 	"github.com/vmware/govmomi/vim25/soap"
 	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/trace"
 )
 
 const (
 	maxBackoffFactor = int64(16)
+
+	VMContextObjectKey = "VMContextObject"
+)
+
+var (
+	m             sync.RWMutex
+	errorHandlers []ErrorHandler
 )
 
+type ErrorHandler interface {
+	// Handle returns true if the error is expected. Otherwise, return false. If any error happens during error handling, return (true, error).
+	Handle(ctx context.Context, err error) (bool, error)
+}
+
+type TaskInProgressHandler struct{}
+
+func (h *TaskInProgressHandler) Handle(ctx context.Context, err error) (bool, error) {
+	if isTaskInProgress(err) {
+		// TaskInProgress error, no need to fail here
+		log.Debugf("TaskInProgress error, continue")
+		return true, nil
+	}
+	return false, nil
+}
+
+func init() {
+	RegisterErrorHandler(&TaskInProgressHandler{})
+}
+
+func RegisterErrorHandler(handler ErrorHandler) {
```

Comment:
Yes, but you can have conflicts between handlers, where two handlers could behave differently for the same error.

Maybe we need to add another function that does take in a handler, and have the current function call this new one with some default error handlers. This would maintain the current behavior, allowing us to add more "default" handlers, while adding an option to add additional handlers per call.